pkg/apis/crd/v1alpha1/types.go

```diff
@@ -591,3 +596,24 @@ type TierList struct {
 
 	Items []Tier `json:"items"`
 }
+
+// PeerServiceAccounts can use NamespacedName, ObjectSelector or
+// NamespacedObjectSelector to select ServiceAccounts.
+type PeerServiceAccounts struct {
```

Comment:
If we want to define generic structs that can apply to all types of objects, then even this one can be generic too, and we need not to define a struct just for ServiceAccounts?

Reply:
Yes, we could. I think maybe we should change this to generic in another PR for example in our API design refactor PR which we may need this generic struct in another place?

Reply:
No strong opinion. But if we are sure about the direction, probably better to make the change now.

Reply:
OK, do you think `PeerObjects` is a good name?

Reply:
Humm..

`EntitySet`? No good idea.

Reply:
Doesn't have this struct now. Resolve this conversation. We could discuss the generic struct in our API refactor PR.